src/WebJobs.Extensions.SendGrid/Config/ISendGridClientFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Extensions.Clients;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Config
+{
+    internal interface ISendGridClientFactory
```

Comment:
Also, does the folder name "Clients" mean you're going to have multiple, or should it just be "Client"?